scipy/optimize/tests/test_optimize.py

```diff
@@ -2180,6 +2181,33 @@ def func(z, *params):
                           match=r'Either final optimization did not succeed'):
             optimize.brute(func, self.rranges, args=self.params, disp=True)
 
+        message = "Polishing is incompatible with integrality constraints"
+        with pytest.warns(OptimizeWarning, match=message):
+            optimize.brute(func, self.rranges, args=self.params,
+                           integrality=True)
+
```

Comment:
No test for what happens if `(high, low)` is provided for an integrality constraint.

Reply:
For this and the comments below, I would always do this sort of thing in new code. I think you'll find this sort of attention to detail very consistent in most of my other PRs. While working on this, however, the mindset was that I was trying to teach an old function a new trick but not reform its ways. There is none of this sort of protection in `brute` already, and I didn't want to get in the business of overhauling it. For example, `brute` currently has no protection against:
`brute(f, [(10, 0), (0, 10)])`

Reply:
Let's see what `brute` does in this situation in `main`

```python3
import numpy as np
from scipy.optimize import brute, rosen

def f(x):
    f.min = min(f.min, x[1])
    f.max = max(f.min, x[1])
    return rosen(x)
f.min = np.inf
f.max = -np.inf

res = brute(f, [(10, 20), (20, 10)], finish=False)
print(f"Solution: {res}\n"
      f"Range of x[1] explored: {(f.min, f.max)}")
```

It doesn't treat `ranges` as the other global optimizers treat `bounds`.
In fact, it doesn't think of `ranges` as `bounds` at all - it doesn't respect them with a callable `finish` (default).

```python3
res = brute(f, [(10, 20), (20, 10)])
print(f"Solution: {res}\n"
      f"Range of x[1] explored: {(f.min, f.max)}")
```
Prints
```
Solution: [0.99999553 0.99999105]
Range of x[1] explored: (0.9799755836218369, 156.79687621398614)
````

Reply:
The new strategy for integers treats `ranges` as `bounds`, so I don't think it's compatible with how `brute` worked before.